modules/javafx.graphics/src/main/native-glass/gtk/glass_window.cpp

```diff
@@ -818,25 +884,38 @@ WindowContextTop::request_frame_extents() {
     }
 }
 
-void WindowContextTop::activate_window() {
-    Display *display = GDK_DISPLAY_XDISPLAY (gdk_window_get_display (gdk_window));
-    Atom navAtom = XInternAtom(display, "_NET_ACTIVE_WINDOW", True);
-    if (navAtom != None) {
-        XClientMessageEvent clientMessage;
-        memset(&clientMessage, 0, sizeof(clientMessage));
+void WindowContextTop::update_frame_extents() {
+    int top, left, bottom, right;
 
-        clientMessage.type = ClientMessage;
-        clientMessage.window = GDK_WINDOW_XID(gdk_window);
-        clientMessage.message_type = navAtom;
-        clientMessage.format = 32;
-        clientMessage.data.l[0] = 1;
-        clientMessage.data.l[1] = gdk_x11_get_server_time(gdk_window);
-        clientMessage.data.l[2] = 0;
+    if (get_frame_extents_property(&top, &left, &bottom, &right)) {
+        if ((top + right + bottom + left) != 0) {
```

Comment:
Are we 100% sure the extents can never be negative values?

Reply:
I'm 99.99% sure :)